testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/firsttime/PreloadMiniAppWindow.kt

```diff
@@ -1,30 +1,25 @@
 package com.rakuten.tech.mobile.testapp.ui.display.firsttime
 
 import android.annotation.SuppressLint
-import android.app.Activity
 import android.content.Context
 import android.content.SharedPreferences
 import android.net.Uri
 import android.view.LayoutInflater
 import android.view.View
-import android.widget.Button
 import android.widget.TextView
 import androidx.appcompat.app.AlertDialog
 import com.rakuten.tech.mobile.miniapp.MiniAppInfo
 import com.rakuten.tech.mobile.miniapp.testapp.R
 import com.rakuten.tech.mobile.testapp.helper.setIcon
 import java.lang.Exception
 
-class FirstTimeWindow(
-    private val activity: Activity,
-    private val firstTimeLaunchListener: FirstTimeLaunchListener
-) {
+class PreloadMiniAppWindow(private val context: Context, private val preloadMiniAppLaunchListener: PreloadMiniAppLaunchListener) {
```

Comment:
Because that dialog is not shown only in first time, when there is permission change -> it will be shown again. Okay, I will rename other places too.

Reply:
it is user fault if they pass the `ApplicationContext` to add a view in a current opening activity.

Reply:
We can help users to avoid passing `ApplicationContext` using `val activity: Activity`.

Reply:
`val activity: Activity` is okay to me.

Reply:
Put `val activity: Activity` is not a good option. The reason why you caused crash with `ApplicationContext` is you were using it wrongly, not because passing `ApplicationContext`.

Restriction with `Activity` does not help anything here and [this](https://nhancv.medium.com/android-show-dialog-without-activity-context-94661d48400f) can clarify your concern.

Reply:
Thanks for researching it, please proceed with the best solution. I have raised the concern to avoid the crash, thanks.

Reply:
What is the best solution you mentioned?

Reply:
Yeah, I can say, according to the [link you have shared](https://nhancv.medium.com/android-show-dialog-without-activity-context-94661d48400f) there are several ways to pass a "non-dead" context, It depends on what we need; one use-case they mention: "show dialog after own activity finish, call dialog in first activity and want to show it in the next activity also." But, I am not sure what is the current plan (use-case). So, if you are comfortable with your current change as the best solution for your plan then please proceed with it.

What I can suggest in a conclusion, not to use any static context to avoid memory leak.

Reply:
We don't use any static context here in sample app, I just pass the activity exactly what you did.
Because you bring up the `ApplicationContext` issue so I just want to explain that passing `ApplicationContext` is not the problem, we can pass anything we want as long as it meets the purpose and works properly in android.

You can see that Google design the `AlertDialog` with `Context` generally, not `Activity` restriction and your `View` is `AlertDialog`.